src/main/scala/org/broadinstitute/dsde/firecloud/model/ModelSchema.scala

```diff
@@ -14,35 +13,110 @@ import org.broadinstitute.dsde.firecloud.FireCloudConfig
     }
   }
+}
+
+
 
-  def isCollectionType(entityType: String): Try[Boolean] = {
-    getTypeSchema(entityType).map(_.memberType.isDefined)
+object ModelSchemaRegistry {
+  // add new schema types from most specific to most general
+  val schemas: Map[SchemaTypes.SchemaType, ModelSchema] = Map(SchemaTypes.FIRECLOUD -> FirecloudModelSchema, SchemaTypes.FLEXIBLE -> FlexibleModelSchema)
+
+  def getModelForSchemaType(schemaType: SchemaTypes.SchemaType): ModelSchema = schemas.getOrElse(schemaType, schemas.last._2)
+
+  def getModelForEntityType(entityType: String): ModelSchema = {
+    val possibleMatch = schemas.values.collectFirst({case schema if schema.isEntityTypeInSchema(entityType) => schema})
+    possibleMatch.getOrElse(schemas.last._2)
   }
+}
 
-  def getCollectionMemberType(collectionEntityType: String): Try[Option[String]] = {
-    getTypeSchema(collectionEntityType).map(_.memberType)
+
+object FlexibleModelSchema extends ModelSchema {
+
+  def memberTypeFromEntityType(entityType: String): Try[Option[String]] = {
+    Success(Some(entityType.replace("_set", "")).filter(_ => isCollectionType(entityType)))
   }
 
-  def getPlural(entityType: String): Try[String] = {
-    getTypeSchema(entityType).map(_.plural)
+  def isCollectionType(entityType: String): Boolean = {
+    entityType.endsWith("_set")
   }
 
   def getRequiredAttributes(entityType: String): Try[Map[String, String]] = {
-    getTypeSchema(entityType).map(_.requiredAttributes)
+    Success(Map.empty)
+  }
+
+  def getPlural(entityType: String): Try[String] =  {
+    Success(pluralize(entityType))
+  }
+
+  private def pluralize(entityType: String): String =  {
+    entityType + "s"
   }
+
+  def getTypeSchema(entityType: String): Try[EntityMetadata] = {
+    Success(EntityMetadata(pluralize(entityType), Map.empty, Some(entityType+"_members").filter(_ => isCollectionType(entityType))))
+  }
+
+  def supportsBackwardsCompatibleIds(): Boolean = false
 }
 
-object EntityTypes {
-  val types : Map[String, EntityMetadata] = ModelJsonProtocol.impModelSchema.read(
-    Source.fromURL(getClass.getResource(FireCloudConfig.Rawls.model)).mkString.parseJson ).schema
+object FirecloudModelSchema extends ModelSchema {
+
+  object EntityTypes {
+    val types : Map[String, EntityMetadata] = ModelJsonProtocol.impModelSchema.read(
+      Source.fromURL(getClass.getResource(FireCloudConfig.Rawls.model)).mkString.parseJson ).schema
+  }
+
+  def getTypeSchema(entityType: String): Try[EntityMetadata] = {
+    EntityTypes.types.get(entityType) map (Success(_)) getOrElse Failure(new RuntimeException("Unknown entity type: " + entityType))
+  }
+
+  def memberTypeFromEntityType(entityType: String): Try[Option[String]] = {
+    getTypeSchema(entityType).map(_.memberType)
+  }
+
+  def getRequiredAttributes(entityType: String): Try[Map[String, String]] = {
+    getTypeSchema(entityType).map(_.requiredAttributes)
+  }
+
+  def isCollectionType(entityType: String): Boolean = {
```

Comment:
I'm still uncertain about this not returning a `Try`. This used to use `getTypeSchema` so  `Failure` was a possible result. I can't convince myself that the places that use `isCollectionType` could possibly return an "Unknown entity type" error like they used to. Is there evidence that this is a safe change?

Reply:
before my changes it was used in 2 places in the code and both did a .getOrElse(false)